onnxruntime/core/providers/cpu/math/element_wise_ops.h

```diff
@@ -648,11 +762,74 @@ void BroadcastLoopSpan(TBroadcaster& bc, Output& output, Input0Scalar input0scal
 }
 
 template <typename TInput, typename TOutput, typename Input0Scalar, typename Input1Scalar, typename General>
-Status BroadcastTwo(OpKernelContext& context, Input0Scalar input0scalar, Input1Scalar input1scalar, General general) {
-  TBroadcaster<TInput, TInput> bc(*context.Input<Tensor>(0), *context.Input<Tensor>(1));
-  TBroadcastOutput<TOutput> output(bc.GetSpanSize(), *context.Output(0, bc.GetOutputShape()));
-  BroadcastLoop(bc, output, input0scalar, input1scalar, general);
+void BroadcastOneSpan(concurrency::ThreadPool* tp, double unit_cost, TOutput* output_ptr, int64_t output_size, 
+                    const TInput* input0_ptr, int64_t input0_size, const TInput* input1_ptr, int64_t input1_size,
+                             Input0Scalar input0scalar, Input1Scalar input1scalar, General general) {
+  if (input0_size == 1) {
+    concurrency::ThreadPool::TryParallelFor(tp, output_size, 
```

Comment:
Would it be correct to assert(output_size==input1_size) here (and corresponding checks that the sizes match on the other branches below)?

Reply:
yes. added.